Update docs for Emacs editor (#932)
hashicorp/terraform-ls: docs/USAGE.md

## Emacs

> If you are using `use-package`, you can use something like this:

Author: > We should assume that even users who have not upgraded yet may land here

IIUC, MELPA's builds is not immutable. And it keeps replacing the older builds. So I don't the the older documentation is exactly right. I will push a fix where I will instead link the official documentation installation section.

> Would you mind linking to some docs about use-package?

I'm linking the official docs which has more details about use-package: https://emacs-lsp.github.io/lsp-mode/page/installation/#use-package

>  Can you clarify where does the config go exactly and/or link to some relevant docs?

It goes in the `init.el` file which is documented in the official doc. I can add it here, if you think it will help.

> If you are using `use-package`, you can use something like this:

Reviewer: Thanks - this is helpful.

I think you accidentally removed the example config too though?

> It goes in the init.el file which is documented in the official doc. I can add it here, if you think it will help.

The only mention of `init.el` I can find is https://emacs-lsp.github.io/lsp-mode/page/installation/#doom-emacs which I don't think is relevant in this particular context - so yes - I think it would be helpful to document that too, and/or link to some relevant docs upstream.

> If you are using `use-package`, you can use something like this:

Author: I have updated it, can you give it a look ? Thanks!

``` emacs-lisp
(use-package lsp-mode
  :ensure t
  :hook ((terraform-mode . lsp-deferred)))
```

> The above code installs `lsp-mode` and also enables it for the
> terraform-mode.

Reviewer: ```suggestion
```

> The terraform client for lsp-mode supports various features like
> semantic tokens, code lens for references etc. There is more detailed
> documentation [here](https://emacs-lsp.github.io/lsp-mode/page/lsp-terraform-ls/).

Reviewer: ```suggestion
The `lsp-mode` language client for Terraform supports various features, such as
semantic tokens, code lens for references etc. There is more detailed
documentation [here](https://emacs-lsp.github.io/lsp-mode/page/lsp-terraform-ls/).
```
As far as I can tell emacs has a single language client implementation with different configurations, so I think this wording may better reflect that?

Author: There are two implementations available at Emacs side for language server:

- lsp-mode
- elgot

I believe the current wording is more clear, but I can change to the suggested one if you think that's more clear.

Author: > We can omit lsp-mod if the statement is true for both lsp-mode and elgot 

I'm not sure if the statement is true for `elgot`. From a quick look at the docs, it seems nobody has tested it for terraform-ls: https://github.com/joaotavora/eglot#connecting-to-a-server

That being said, even if it works - it will have very minimal features compared to `lsp-mode` as I made sure to support additional features for it - code lens, provides & module calls widget support etc specifically in `lsp-mode`.

>  but what I primarily meant above is that we shouldn't imply there's a "Terraform client" 

Ah, okay. I have made the suggested changes.
